packages/services/src/providers/perps/hyperliquid-perps.provider.ts

```diff
@@ -0,0 +1,741 @@
+      const effectiveInitialCapital = initialCapital ?? totalEquityValue;
+
+      const summary: PerpsAccountSummary = {
+        // Core metrics
+        totalEquity: totalEquityValue,
+        initialCapital: effectiveInitialCapital,
+        availableBalance: totalRawUsd.toNumber(),
+        marginUsed: totalMarginUsed.toNumber(),
+
+        // PnL metrics
+        totalPnl: totalUnrealizedPnl.plus(closedPnl).toNumber(),
+        totalRealizedPnl: closedPnl.toNumber(),
+        totalUnrealizedPnl: totalUnrealizedPnl.toNumber(),
+
+        // Trading statistics
+        totalVolume: totalVolume.toNumber(),
+        totalTrades: tradeCount,
+        totalFeesPaid: totalFees.toNumber(),
+
+        // Position counts
+        openPositionsCount,
+        closedPositionsCount: 0, // Would need historical data
+        liquidatedPositionsCount: 0, // Would need to track liquidation events
+
+        // Performance metrics
+        // Calculate ROI based on provided initial capital
+        roi:
+          effectiveInitialCapital > 0
+            ? ((totalEquityValue - effectiveInitialCapital) /
+                effectiveInitialCapital) *
+              100
+            : 0,
+        roiPercent:
+          effectiveInitialCapital > 0
+            ? ((totalEquityValue - effectiveInitialCapital) /
+                effectiveInitialCapital) *
+              100
+            : 0,
+        averageTradeSize:
+          tradeCount > 0 ? totalVolume.dividedBy(tradeCount).toNumber() : 0,
+
+        // Status
+        accountStatus: openPositionsCount > 0 ? "active" : "inactive",
+
+        // Raw data storage: Only store for sampled requests (1% by default)
+        rawData: undefined,
+      };
+
+      // Sampling for raw data storage
+      const shouldSample = Math.random() < this.SAMPLING_RATE;
+
+      if (shouldSample) {
+        // Store raw data for debugging
+        summary.rawData = { clearinghouseState, recentFills };
+
+        // Also send to Sentry for monitoring
+        Sentry.captureMessage("Hyperliquid API Response Sample", {
+          level: "debug",
+          extra: {
+            response: { clearinghouseState, recentFills },
+            walletAddress: maskedAddress,
+            processingTime: Date.now() - startTime,
+          },
+        });
+
+        this.logger.debug(
+          `[HyperliquidProvider] Sampled request - storing raw data for ${maskedAddress}`,
+        );
+      }
+
+      this.logger.debug(
+        `[HyperliquidProvider] Fetched account summary for ${maskedAddress} in ${Date.now() - startTime}ms`,
+      );
+
+      return summary;
+    } catch (error) {
+      Sentry.captureException(error, {
+        extra: {
+          walletAddress: maskedAddress,
+          method: "getAccountSummary",
+          processingTime: Date.now() - startTime,
+        },
+      });
+
+      this.logger.error(
+        `[HyperliquidProvider] Error fetching account summary for ${maskedAddress}:`,
+        error,
```

Comment:
[nit] pino expects the object to come _before_ the message, which then makes them available as metadata in our log service:
```ts
this.logger.error(
  {  
      error,
      address: maskedAddress
   },
  `[HyperliquidProvider] Error fetching account summary`
)
```